pkg/controllers/cspi-controller/handler.go

```diff
@@ -195,40 +195,26 @@ func (c *CStorPoolInstanceController) updateStatus(cspi *cstor.CStorPoolInstance
 	var status cstor.CStorPoolInstanceStatus
 	var err error
 	pool := zpool.PoolName()
+	propertyList := []string{"health", "io.openebs:readonly"}
 
-	state, er := zpool.GetPropertyValue(pool, "health")
-	if er != nil {
-		err = zpool.ErrorWrapf(err, "Failed to fetch health")
+	valueList, er := zpool.GetListOfPropertyValues(pool, propertyList)
+	if er != nil || len(valueList) != len(propertyList) {
+		return errors.Wrapf(err, "Failed to fetch %v output: %v", propertyList, valueList)
 	} else {
-		status.Phase = cstor.CStorPoolInstancePhase(state)
-	}
-
-	freeSize, er := zpool.GetPoolCapacity(pool, "free")
-	if er != nil {
-		err = zpool.ErrorWrapf(err, "Failed to fetch free size")
-	} else {
-		status.Capacity.Free = freeSize
-	}
-
-	usedSize, er := zpool.GetPoolCapacity(pool, "allocated")
-	if er != nil {
-		err = zpool.ErrorWrapf(err, "Failed to fetch used size")
-	} else {
-		status.Capacity.Used = usedSize
-	}
-
-	totalSize, er := zpool.GetPoolCapacity(pool, "size")
-	if er != nil {
-		err = zpool.ErrorWrapf(err, "Failed to fetch total size")
-	} else {
-		status.Capacity.Total = totalSize
+		status.Phase = cstor.CStorPoolInstancePhase(valueList[0])
+		if valueList[1] == "on" {
+			status.ReadOnly = true
+		}
```

Comment:
What are the magical numbers `0` and `1` ? 
If we need to use them -- a little comment can help understand.